### DISCUSSION TITLE
where is pre-checker component

### DISCUSSION THREAD

**Discussion by jschmi102** [joomla/joomla-cms#34678] [Q&A]
hi,
I updated my developer site to 3.10 apha 8. This worked without any problems and my site runs fine with 3.10.  Now I tried to use the pre-checker component to check intstalled extensions but I cannot find it. Any clue what is wrong?

**Comment by Bakual** [comment]
You need to set the Joomla Update options to this settings:
![image](https://user-images.githubusercontent.com/1018684/124119109-6163f600-da72-11eb-91ed-22edc8d3d45f.png)

Then you see the pre-checker.

**Comment by jschmi102** [comment]
thanks a lot - found it...

**Comment by jschmi102** [comment]
hi,
There is still some confusion with the pre-checker:
E.g- I'm using for my content-plugin a update server at all times succesfully. The pre-checker instead says that there is "No Compatibility Information" 
I've definen in plugin xml:
```
	<updateservers>
		<server type="extension" priority="1" name="rgraph Updates">http://www.jschmidt-systemberatung.de/downloads/rgraph_v2_updates.xml</server>
	</updateservers>
```
and in
rgraph_v2_updates.xml:
```
	<update>
		<name>rgraph Content plugin</name>
		<description>rgraph Content plugin</description>
		<element>rgraph</element>
		<type>plugin</type>
		<client>0</client>
		<folder>content</folder>
		<version>2.1.4.2</version>
		<infourl title="rgraph">http://www.jschmidt-systemberatung.de/</infourl>
		<downloads>
			<downloadurl type="full" format="zip">http://www.jschmidt-systemberatung.de/downloads/rgraph4Joomla-v2.1.4.zip</downloadurl>
		</downloads>
		<tags>
			<tag>stable</tag>
		</tags>
		<targetplatform name="joomla" version="3.10" />
	</update>
	<update>
		<name>rgraph Content plugin</name>
		<description>rgraph Content plugin</description>
		<element>rgraph</element>
		<type>plugin</type>
		<client>0</client>
		<folder>content</folder>
		<version>2.1.4.2</version>
		<infourl title="rgraph">http://www.jschmidt-systemberatung.de/</infourl>
		<downloads>
			<downloadurl type="full" format="zip">http://www.jschmidt-systemberatung.de/downloads/rgraph4Joomla-v2.1.4.zip</downloadurl>
		</downloads>
		<tags>
			<tag>stable</tag>
		</tags>
		<targetplatform name="joomla" version="4.[0]" />
	</update>
```
are there naming problems in pre-checker? Or what goes wrong?

**Reply by Bakual** [reply]
The 3.10 checker doesn't check for 4.0 yet, since 4.0 isn't released. So it will check the 3.10 part. And there I think you need to adjust the targetplatform to `3.(10)` or so. For my component `3.([89]|10)` was working.
Also I'm not sure if you need to have the client tag written as `site` and `administrator` instead of `0` and `1`. I haven't tested that. Maybe @zero-24 knows why there are differences between the regular update and the checker (I think those differences are bad).

**Comment by zero-24** [comment]
re @Bakual 
> The 3.10 checker doesn't check for 4.0 yet, since 4.0 isn't released.

That is not true. but you have to choose an update server that points to 4.x. ;)

> So it will check the 3.10 part.

Yes depending on the update server settings.

>  And there I think you need to adjust the targetplatform to 3.(10) or so. For my component 3.([89]|10) was working.

Yes as long as the regex matches all supported joomla versions 3.10 works too.

> Also I'm not sure if you need to have the client tag written as site and administrator instead of 0 and 1.

That fallback has been removed from 4.0 and should not be used in 3.x anyway.

> Maybe @zero-24 knows why there are differences between the regular update and the checker (I think those differences are bad).

I'm not aware of specific difference cause we are using the same libaray code. The only thing I'm aware of is the minimum stability tag where there is an Issue open already.

re @jschmi102 
> <targetplatform name="joomla" version="4.[0]" />

Please try `<targetplatform name="joomla" version="4.0" />` that works for me.

Please also add the "folder" attribute as it is required for plugins to the update XML as well as changing the client to "site".

In order to check against 4.0 please follow the instructions listed here: 
- Switch the Update Channel to “Custom URL” and set the Custom URL to “https://update.joomla.org/core/test/310to4_list.xml”
- Now you get, updated to the latest 4.0 Release Candidate and the pre upgrade checker checks against 4.x

**Comment by jschmi102** [comment]
hi,
thanks for your response!!
Since I changed the targetplatform to `version="3.(9|10")` and `version="4.[0]" ` Info of pre-checker seems to be OK (updater of joomla4 and joomla3 do work with this  as well)
I've installed an extension which uses a generic definition: `<targetplatform name="joomla" version=".*"/> `
this leads to an info of comaptibility for 3.x and 4.x - do you think this to be correctly checked?

**Reply by zero-24** [reply]
Yes cause .* matches both 3.x and 4.x thats the reason this should not be used cause you never know whether your extension is compatible with an upcomming joomla version for example joomla 5.0 or 6.0

**Comment by jschmi102** [comment]
hi,
is there any  difference (now and for future versions) in processing,,  if you specify `<client>0</client>` or `<client>site</client>`
Docs state that "0" is translated to "site"

**Reply by zero-24** [reply]
0 only translate to site in 3.x this is a b/c behavior thats finally removed from 4.x

**Reply by jschmi102** [reply]
hi,
ran updates a couple times with joomla4 beta and joomls4 RC - but did not see any problem with my setting ("0"). Is it ignored by joomla 4 or defaulted if "wrong" value used?
